portfolio/src/main/webapp/script.js

```diff
@@ -47,4 +47,18 @@ function randomPic() {
   picElem.src = pics[randomFromArray(pics)]
   //make image visible
   picElem.style.display = "block"
+}
+
+/**
+ * Do hello is a dynamic approach to the /hello servlet
+ */
+async function doHello() {
+    //gets text from servlet
```

Comment:
FYI: You can skip these comments. They don't add any value. It is very clear from the code that you are fetching from the server.  The general readability recommendation is to add comments only if the code is not clear.

You can leave it as it is, since I am not strongly enforcing it for SPS, but it is strongly enforced at Google.